x/evm/keeper/grpc_query.go

```diff
@@ -313,10 +312,15 @@ func (k Keeper) EstimateGas(c context.Context, req *txs.EthCallRequest) (*txs.Es
 		hi = req.GasCap
 	}
 	txMsg := args.ToTransaction()
-	// set aspect tx context
-	ethTxContext := artelatypes.NewEthTxContext(txMsg.AsTransaction())
-	k.GetAspectRuntimeContext().SetEthTxContext(ethTxContext)
-	k.GetAspectRuntimeContext().WithCosmosContext(ctx)
+
+	// Aspect Runtime Context Lifecycle: create aspect context.
+	// This marks the beginning of running an aspect of EstimateGas, creating the aspect context,
+	// and establishing the link with the SDK context.
+	ctx, aspectCtx := k.WithAspectContext(ctx, txMsg.AsTransaction())
+	defer func() {
+		ctx = ctx.WithValue(artelatypes.AspectContextKey, nil)
```

Comment:
Same as above here, this will not destroy the Aspect Context in the Context, instead it creates new Context with `AspectContextKey` set to nil. I think it's better to create a map or something and save it to the ctx value, then in the end we could delete the specific map key to ensure that the context gets cleared.

Reply:
@dumbeng fixed.